### PR TITLE
feat(workflow): Filter out performance alerts and rules if they don't have the feature

### PR DESCRIPTION
**Pull request by iProgramStuff**
If a user had access to performance alerts, created some rules and had some alerts fire but then downgrade, we will hide those alerts from the UI. We accomplish this by filtering them out of the endpoints.

This is mostly serving as a stop-gap - the longer-term plan is to actually "disable" rules when they've downgraded and have some FE/UI support for this.

### CODE REVIEW

File: src/sentry/incidents/endpoints/project_alert_rule_index.py [hunk 1]
```diff
@@ -21,14 +21,19 @@ def get(self, request, project):
         """
         Fetches alert rules and legacy rules for an organization
         """
+        alert_rules = AlertRule.objects.fetch_for_project(project)
+        if not features.has("organizations:incidents-performance", project.organization):
+            # Filter to only error alert rules
+            alert_rules = alert_rules.filter(snuba_query__dataset="events")
```
Comment (wedamija) [inline src/sentry/incidents/endpoints/project_alert_rule_index.py:27]: Should use `sentry.snuba.dataset.Dataset.Events` here and above.

**Review by wedamija — COMMENTED** [review]
So later on will we actually show these rules/alerts but with the disabled state instead?

**Comment by iProgramStuff** [comment]
Not if they don't have the feature, it seems.

In the future, I think when a user can manually disable/enable alerts, then we would show those.

That's why I opted to filter based on the feature rather than the status of the alert rule (I considered this) - eventually we'll want to show disabled alerts that they have access to.

File: src/sentry/incidents/endpoints/organization_incident_index.py [hunk 2]
```diff
@@ -42,6 +43,10 @@ def get(self, request, organization):
             elif query_status == "closed":
                 incidents = incidents.filter(status=IncidentStatus.CLOSED.value)
 
+        if not features.has("organizations:incidents-performance", organization):
+            # Filter to only error alerts
+            incidents = incidents.filter(alert_rule__snuba_query__dataset=Dataset.Events)
```
Comment (wedamija) [inline src/sentry/incidents/endpoints/organization_incident_index.py:48]: You'll need `Dataset.Events.value` here to get the string.

File: src/sentry/incidents/endpoints/project_alert_rule_index.py [hunk 3]
```diff
@@ -21,14 +22,19 @@ def get(self, request, project):
         """
         Fetches alert rules and legacy rules for an organization
         """
+        alert_rules = AlertRule.objects.fetch_for_project(project)
+        if not features.has("organizations:incidents-performance", project.organization):
+            # Filter to only error alert rules
+            alert_rules = alert_rules.filter(snuba_query__dataset=Dataset.Events)
```
Comment (wedamija) [inline src/sentry/incidents/endpoints/project_alert_rule_index.py:28]: Same as above with the `.value`

File: tests/sentry/incidents/endpoints/test_project_alert_rule_index.py [hunk 4]
```diff
@@ -38,7 +38,7 @@ def test_simple(self):
         alert_rule = self.create_alert_rule()
 
         self.login_as(self.user)
-        with self.feature("organizations:incidents"):
+        with self.feature(["organizations:incidents", "organizations:incidents-performance"]):
```
Comment (wedamija) [inline tests/sentry/incidents/endpoints/test_project_alert_rule_index.py:41]: Worth adding a test to verify that the alert rule doesn't appear if you don't have performance?

**Review by wedamija — APPROVED** [review]

**Pull request merged**